[node-manager][docs] Improved NodeGroupConfiguration documentation (#6514)
deckhouse/deckhouse: modules/040-node-manager/docs/README.md

- обновления ядра ОС на конкретную версию ([пример](faq.html#как-обновить-ядро-на-узлах));
- изменение параметров ОС ([пример настройки параметра sysctl](examples.html#задание-параметра-sysctl));
- сбор информации на узле и выполнение других подобных действий. 

Ресурс `NodeGroupConfiguration` позволяет указывать [приоритет](cr.html#nodegroupconfiguration-v1alpha1-spec-weight) выполняемым скриптам, ограничивать их выполнение определенными [группами узлов](cr.html#nodegroupconfiguration-v1alpha1-spec-nodegroups) и [типами ОС](cr.html#nodegroupconfiguration-v1alpha1-spec-bundles).

Код скрипта указывается в параметре [content](cr.html#nodegroupconfiguration-v1alpha1-spec-content) ресурса. При создании скрипта на узле содержимое параметра `content` проходит через шаблонизатор [Go Template](https://pkg.go.dev/text/template), который позволят встроить дополнительный уровень логики при генерации скрипта. При прохождении через шаблонизатор становится доступным контекст с набором динамических переменных.

> Переменные, которые доступны для использования в шаблонизаторе: 

Reviewer: ```suggestion
The following variables are supported by the templating engine: 
```

<ul>

> <li><code>.cloudProvider</code> (для групп узлов с nodeType <code>CloudEphemeral</code> или <code>CloudPermanent</code>) — массив данных облачного провайдера.
> {% offtopic title="Пример данных..." %}

Reviewer: ```suggestion
<li><code>.cloudProvider</code> (for node groups of nodeType <code>CloudEphemeral</code> or <code>CloudPermanent</code>) — cloud provider data.
{% offtopic title="Example of data..." %}
```

```yaml
cloudProvider:
  instanceClassKind: OpenStackInstanceClass
  machineClassKind: OpenStackMachineClass
  openstack:
    connection:
      authURL: https://cloud.provider.com/v3/
      domainName: Default
      password: p@ssw0rd
      region: region2
      tenantName: mytenantname
      username: mytenantusername
    externalNetworkNames:
    - public
    instances:
      imageName: ubuntu-22-04-cloud-amd64
      mainNetwork: kube
      securityGroups:
      - kube
      sshKeyPairName: kube
    internalNetworkNames:
    - kube
    podNetworkMode: DirectRoutingWithPortSecurityEnabled
  region: region2
  type: openstack
  zones:
  - nova
```
{% endofftopic %}</li>

> <li><code>.cri</code> — используемый CRI (с версии Deckhouse 1.49 используется только <code>Containerd</code>).</li>
> <li><code>.kubernetesVersion</code> — используемая версия Kubernetes.</li>
> <li><code>.nodeUsers</code> — массив данных группы узлов.
> {% offtopic title="Пример данных..." %}

Reviewer: ```suggestion
<li><code>.cri</code> — the CRI in use (starting with Deckhouse 1.49, only <code>Containerd</code> is supported).</li>
<li><code>.kubernetesVersion</code> — the Kubernetes version in use.</li>
<li><code>.nodeUsers</code> — node group data.
{% offtopic title="Example of data..." %}
```

Reviewer: Got to be **_node user_** data here, given that the node group data is below 🤔  By the way, what's **_массив_** in "массив данных"?

```yaml
nodeUsers:
- name: user1
  spec:
    isSudoer: true
    nodeGroups:
    - '*'
    passwordHash: PASSWORD_HASH
    sshPublicKey: SSH_PUBLIC_KEY
    uid: 1050
```
{% endofftopic %}
</li>

> <li><code>.nodeGroup</code> — массив данных группы узлов.
> {% offtopic title="Пример данных..." %}

Reviewer: ```suggestion
<li><code>.nodeGroup</code> — node group data.
{% offtopic title="Example of data..." %}
```
